typegate/src/planner/mod.ts

```diff
@@ -470,4 +493,35 @@ export class Planner {
   private formatPath(path: string[]) {
     return [this.operationName, ...path].join(".");
   }
+
+  /**
+   * Idea:
+   * query: [a1, a2, ..], mutation: [b1, a1, b2, ..]
+   * => a1: [query, mutation], a2: [query], ..., b1: [mutation] ...
+   */
+  private getReverseMapNameToQuery() {
+    const indices = ["query", "mutation"].map((label) =>
+      this.tg.type(0, Type.OBJECT).properties?.[label]
+    );
+    const res = new Map<string, Set<string>>();
+    for (const idx of indices) {
+      const { fields, title } = this.collectFieldNames(idx);
```

Comment:
unless the function can be used elsewhere, I would inline it to avoid confusion with the usage of the function (and construction object to deconstruct it afterwards)

Reply:
`getReverseMapNameToQuery` calls `collectFieldNames`, both only depend on `tg: Typegraph` so we can generalize, and move the definitions to `utils.ts` or inline above the class definition.